### PR TITLE
Add support for passing line and column number to ConsoleLauncher via `--select-file` and `--select-resource`

### LINKED ISSUE #3483: Add support for passing line and column number to ConsoleLauncher via `--select-file` and `--select-resource`
This should be done using the same syntax that is supported for the URI argument of dynamic tests (`?line=...&column=...`).

## Related issues
- https://github.com/junit-team/junit5/issues/3366#issuecomment-1625211213
- #3484
- #3485

### PR DESCRIPTION
**Pull request by juliette-derancourt**
## Overview

Resolves #3483 

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [ ] [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [ ] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [ ] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [ ] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)

### CODE REVIEW

File: junit-platform-console/src/main/java/org/junit/platform/console/options/SelectorConverter.java [hunk 1]
```diff
@@ -109,4 +120,12 @@ public DiscoverySelectorIdentifier convert(String value) {
 		}
 	}
 
+	private static URI stripQueryComponent(URI uri) {
```
Comment (juliette-derancourt) [inline junit-platform-console/src/main/java/org/junit/platform/console/options/SelectorConverter.java:123]: This method is a duplicate of the one in `org.junit.platform.engine.support.descriptor.ResourceUtils`, but I don't think there is a way of using the same in both places, is it?

Comment (marcphilipp) [reply junit-platform-console/src/main/java/org/junit/platform/console/options/SelectorConverter.java:123]: If we made `ResourceUtils` and the method public, we could reuse it. I think it would be slightly better regarding consistenct to move it to the `org.junit.platform.commons.util` package, though.

**Review by marcphilipp — COMMENTED** [review]
Looking good! :+1:

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/ResourceUtils.java [hunk 2]
```diff
@@ -8,19 +8,21 @@
...
-import org.junit.platform.commons.util.StringUtils;
+import org.apiguardian.api.API;
 
 /**
  * Collection of static utility methods for working with resources.
  *
  * @since 1.3
  */
-final class ResourceUtils {
+@API(status = INTERNAL, since = "1.3")
```
Comment (juliette-derancourt) [inline junit-platform-commons/src/main/java/org/junit/platform/commons/util/ResourceUtils.java:24]: @marcphilipp Should I leave this as "since 1.3" or change it to the current version?

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/ResourceUtils.java [hunk 3]
```diff
@@ -8,19 +8,21 @@
...
 import java.net.URI;
 
-import org.junit.platform.commons.util.Preconditions;
-import org.junit.platform.commons.util.StringUtils;
+import org.apiguardian.api.API;
 
 /**
  * Collection of static utility methods for working with resources.
  *
  * @since 1.3
```
Comment (marcphilipp) [inline junit-platform-commons/src/main/java/org/junit/platform/commons/util/ResourceUtils.java:22]:
```suggestion
 * @since 1.3 (originally in org.junit.platform.engine.support.descriptor)
```

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/ResourceUtils.java (hunk 2 above)
Comment (marcphilipp) [reply junit-platform-commons/src/main/java/org/junit/platform/commons/util/ResourceUtils.java:24]: Since it's now public, we should do:
```suggestion
@API(status = INTERNAL, since = "1.12")
```

**Review by marcphilipp — APPROVED** [review]
